Assets/Scripts/UI/HealthBar.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using RogueApeStudio.Crusader.HealthSystem;
+
+namespace RogueApeStudio.Crusader.UI.Healthbar
+{
+    public class HealthBar : MonoBehaviour
+    {
+        private float _barMaxWidth;
+        private float _barWidth;
+        private float _velocity;
+
+        [SerializeField] private RectTransform _bar;
+        [SerializeField] private Health _player;
+
+        private void Start()
+        {
+            _barMaxWidth = _bar.sizeDelta.x;
+        }
+
+        private void Update()
```

Comment:
Does this need to happen each update, or can this subscribe to the Events in `Health`?
Also, take a peek at DoTweening this, I suspect there is a method available.

Reply:
As a general rule of thumb, unless you _**Absolutely**_ need to do something each frame, a smart use of events and dotweening can often lead to the same outcome.

Reply:
For now I will keep it as the Mathf.SmoothDamp which needs to be in update so that we can start testing the game. we can change this when we start optimizing the game.

Reply:
`There is nothing so permanent as a temporary fix`.
I'll make an issue of it, so you don't forget.